.github/workflows/docker-build.yml

```diff
@@ -0,0 +1,92 @@
+name: Docker build and publish and optional kubernetes deploy
+on:
+  workflow_call:
+    inputs:
+      image-name:
+        required: true
+        type: string
+      image-repo-name:
+        required: false
+        type: string
+        default: "ops"
+      deployment-name:
+        required: false
+        type: string
+      cluster-name:
+        required: false
+        type: string
+      cluster-zone:
+        required: false
+        type: string
+      gcp-project:
+        description: "Google Cloud Platform project id."
+        type: string
+        required: true
+      docker-push:
+        description: "Whether or not to push the docker image."
+        type: boolean
+        required: false
+        default: false
+    secrets:
+      GOOGLE_AUTHENTICATION_CREDENTIALS_JSON:
+        description: "Google Cloud Platform service-agent JSON credentials for accessing our Artifact Repository and installing private packages."
+        required: false
+env:
+  full_image_uri: europe-north1-docker.pkg.dev/${{ inputs.gcp-project }}/${{ inputs.image-repo-name }}/${{ inputs.image-name }}
+jobs:
+  docker_build:
+    runs-on: ubuntu-latest
+    env:
+      DOCKER_BUILDKIT: 1
+    steps:
+      - uses: actions/checkout@v3
+      - name: Authenticate with Google Cloud
+        uses: google-github-actions/auth@v1
+        with:
+          credentials_json: ${{ secrets.GOOGLE_AUTHENTICATION_CREDENTIALS_JSON }}
+      - name: Authorize Docker push
+        run: gcloud auth configure-docker europe-north1-docker.pkg.dev
+      - name: Build builder image
+        run: >-
+          docker build
+          --build-arg BUILDKIT_INLINE_CACHE=1
+          -f Dockerfile
+          --cache-from ${{ env.full_image_uri }}-builder:latest
+          -t ${{ env.full_image_uri }}-builder:latest
+          --target builder
+          .
+      - name: Build runtime image
+        run: >-
+          docker build
+          --build-arg BUILDKIT_INLINE_CACHE=1
+          -f Dockerfile
+          --cache-from ${{ env.full_image_uri }}:latest
+          -t ${{ env.full_image_uri }}:latest
+          -t ${{ env.full_image_uri }}:${{ github.sha }}
+          .
+      - name: Push builder image to Artifact Registry
+        if: ${{ inputs.docker-push }}
+        run: docker push --all-tags ${{ env.full_image_uri }}-builder
+      - name: Push runtime image to Artifact Registry
+        if: ${{ inputs.docker-push }}
+        run: docker push --all-tags ${{ env.full_image_uri }}
+
+  kubernetes_deploy:
```

Comment:
Eventually it might be nice to expand this with a `cloud_run_deploy` but we can leave that for another time :+1:

Reply:
I'll turn this into an issue so we don't forget :+1:

Reply:
Good idea

Reply:
#15